superduperdb/base/datalayer.py

```diff
@@ -1042,23 +1044,39 @@ def replace(
             info = self.metadata.get_component(
                 object.type_id, object.identifier, version=object.version
             )
-        except FileNotFoundError:
+        except FileNotFoundError as e:
             if upsert:
                 return self.apply(
                     object,
                 )
-            raise FileNotFoundError
+            raise e
 
         # If object has no version, update the last version
         object.version = info['version']
 
+        serialized = object.dict().encode(leaves_to_keep=(Component,))
```

Comment:
Since these 2 lines are used to retrieve children's and are used a lot of places 
can we make a helper function in SUperDuperFlatEncode ?

Reply:
Yes.